javaslang/src/main/java/javaslang/collection/CharSeq.java

```diff
@@ -446,36 +445,25 @@ public CharSeq insertAll(int index, Iterable<? extends Character> elements) {
             sb.append(element);
         }
         sb.append(javaString.substring(index));
-        return of(sb.toString());
+        return of(sb);
     }
 
     @Override
     public Iterator<Character> iterator() {
-        return new AbstractIterator<Character>() {
-            private int index = 0;
-
-            @Override
-            public boolean hasNext() {
-                return index < back.length();
-            }
-
-            @Override
-            public Character getNext() {
-                return back.charAt(index++);
-            }
-        };
+        return Iterator.ofAll(toCharArray());
     }
 
     @Override
     public CharSeq intersperse(Character element) {
         final StringBuilder sb = new StringBuilder();
-        for (int i = 0; i < length(); i++) {
-            if (i > 0) {
```

Comment:
this condition is only false for the first statement: let's extract that to simplify and optimize the logic